Client/Frontend/UserContent/UserScripts/MainFrame/AtDocumentStart/TrackingProtectionStats.js

```diff
@@ -4,182 +4,178 @@
-      // We only handle sync requests
-      this._shouldTrack = arguments[2] !== undefined && !arguments[2]
-      this._url = url;
-      return originalOpen.apply(this, arguments);
-    };
-
-    xhrProto.send = function(body) {
-      if (this._url === undefined || !this._shouldTrack) {
-        return originalSend.apply(this, arguments);
+          originalOpen = originalSend = originalImageSrc = mutationObserver = null;
+        }
+        return;
       }
-      
-      // Only attach the `error` event listener once for this
-      // `XMLHttpRequest` instance.
-      if (!this._tpErrorHandler) {
-        // If this `XMLHttpRequest` instance fails to load, we
-        // can assume it has been blocked.
-        this._tpErrorHandler = function() {
-          sendMessage(this._url, "xmlhttprequest");
-        };
-        this.addEventListener("error", this._tpErrorHandler);
+
+      // -------------------------------------------------
+      // Send ajax requests URLs to the host application
+      // -------------------------------------------------
+      var xhrProto = XMLHttpRequest.prototype;
+      if (!originalOpen) {
+        originalOpen = xhrProto.open;
+        originalSend = xhrProto.send;
       }
-      return originalSend.apply(this, arguments);
-    };
-
-    // -------------------------------------------------
-    // Detect when new sources get set on Image and send them to the host application
-    // -------------------------------------------------
-    if (!originalImageSrc) {
-      originalImageSrc = Object.getOwnPropertyDescriptor(Image.prototype, "src");
-    }
-    delete Image.prototype.src;
-    Object.defineProperty(Image.prototype, "src", {
-      get: function() {
-        return originalImageSrc.get.call(this);
-      },
-      set: function(value) {
+
+      xhrProto.open = $(function(method, url) {
+        // Blocked async XMLHttpRequest are handled via RequestBlocking.js
+        // We only handle sync requests
+        this._shouldTrack = arguments[2] !== undefined && !arguments[2]
+        this._url = url;
+        return originalOpen.apply(this, arguments);
+      });
+
+      xhrProto.send = $(function(body) {
+        if (this._url === undefined || !this._shouldTrack) {
+          return originalSend.apply(this, arguments);
+        }
+        
         // Only attach the `error` event listener once for this
-        // Image instance.
+        // `XMLHttpRequest` instance.
         if (!this._tpErrorHandler) {
-          // If this `Image` instance fails to load, we can assume
-          // it has been blocked.
-          this._tpErrorHandler = function() {
-            sendMessage(this.src, "image");
-          };
+          // If this `XMLHttpRequest` instance fails to load, we
+          // can assume it has been blocked.
+          this._tpErrorHandler = $(function() {
+            sendMessage(this._url, "xmlhttprequest");
```

Comment:
This `_url` really gave me a hassle at one point. I used a similar generically named variable in my code and it turned out it interfered with this one. However mine was a URL instead of String (for good reason). And I had no idea why things were not working. Turned out this script was replacing my `_url` (or the other way around). Ideally we would use a) something more strongly typed than a string, b) some name that is specific to this script. But I don't expect you to change this here, it's just a general thought about this sort of stuff.